src/api/secret.rs

```diff
@@ -36,7 +38,131 @@ pub struct SecretAuth {
     #[serde(default)]
     pub entity_id: String,
     #[serde(default)]
-    pub lease_duration: u32,
+    pub lease_duration: u64,
     #[serde(default)]
     pub renewable: bool,
 }
+
+impl Secret {
+    pub fn token_is_renewable(&self) -> Result<bool, RvError> {
+        if let Some(auth) = &self.auth {
+            return Ok(auth.renewable);
+        }
+
+        if let Some(data) = self.data.get("renewable") {
+            if let Some(renewable) = data.as_bool() {
+                return Ok(renewable);
+            } else {
+                return Err(rv_error_string!("token id found but in the wrong format"));
+            }
+        }
+
+        Ok(false)
+    }
+
+    pub fn token_id(&self) -> Result<String, RvError> {
+        if let Some(auth) = &self.auth {
+            if !auth.client_token.is_empty() {
+                return Ok(auth.client_token.clone());
+            }
+        }
+
+        if let Some(data) = self.data.get("id") {
+            if let Some(id) = data.as_str() {
+                return Ok(id.to_string());
+            } else {
+                return Err(rv_error_string!("token id found but in the wrong format"));
+            }
```

Comment:
This can be replaced by something like

~~~
let Some(id) = data.as_str() else {
    return Err(rv_error_string!("token id found but in the wrong format"));
}
return Ok(id.to_string());
~~~